### PR TITLE
fix(): fixed paths internal in files

### PR DESCRIPTION
**Pull request by Brocco**
The paths/url references inside files were not changed to allow for dynamic paths
Added the ability to utlize dynamic paths from within the project structure

### CODE REVIEW

File: addon/ng2/blueprints/component/index.js [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+    }
   },
 
   fileMapTokens: function(options) {
     // Return custom template variables here.
+    var _this = this;
     return {
       __name__: function(options) {
-        return options.locals.name;
+        return _this.dynamicPath.name;
```
Comment (hansl) [inline addon/ng2/blueprints/component/index.js]: Try out arrow functions, it will simplify your code greatly. And they've been in Node for a while now.

Comment (Brocco) [inline addon/ng2/blueprints/component/index.js]: I keep forgetting that... I use them a **lot** when doing TypeScript so when I'm doing JavaScript I think I'm always restricted to ES5 syntax. Good catch, new PR pushed!

**Comment by filipesilva** [comment]
Could you add a few tests in `e2e_workflow.js` to make sure the dynamic paths are working btw?

Also I see some the new blueprints for services, but not the supporting file changes (like `locals` or `fileMapTokens` with defaults in `blueprints/service/index.js`) nor the removal of the old absolute pathed blueprints. Is this intended?

**Comment by Brocco** [comment]
@filipesilva instead of adding tests to the e2e workflow suite, I added a new acceptance test to handle the testing of generating components

**Comment by IgorMinar** [comment]
I haven't reviewed every single line, but at a high level this looks good to me

File: tests/acceptance/generate.component.spec.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
+'use strict';
```
Comment (filipesilva) [inline tests/acceptance/generate.component.spec.js]: A minor thing, but shouldn't the file be named `generate-component.spec.js`? We overall go for dash separated file names, and reserve the dots to separate functionality groups.

**Comment by filipesilva** [comment]
@Brocco tests look good, and now we have appveyor so we can tests on windows as well.

What about the changes to services though, what are they for? Do they also support relative path names? I don't quite understand why these files are also in the PR.

**Comment by Brocco** [comment]
@filipesilva I want to use this a test to see how the path structuring feels after some usage before fully integrating into all other generations. Also, awaiting on a more stabilized best practices project structure.

File: addon/ng2/blueprints/component/index.js [hunk 3]
```diff
@@ -1,26 +1,59 @@
...
+    var rootPath = path.join(this.project.root, 'src', 'app');
+    
+    var outputPath = path.join(rootPath, entityName);
+    
+    if (entityName.indexOf(path.sep) === 0) {
+      outputPath = path.join(rootPath, entityName.substr(1));
+    } else if (cwd.indexOf(rootPath) >= 0) {
+      outputPath = path.join(cwd, entityName);
+    } else if (cwd.indexOf(path.join(this.project.root, 'src')) >= 0
+      && entityName.indexOf('app') === 0) {
```
Comment (hansl) [inline addon/ng2/blueprints/component/index.js]: left aligned inside the `(` of the if.

**Comment by hansl** [comment]
LGTM.

**Pull request closed without merging**